WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,151 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+using Microsoft.IdentityModel.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WinUIMSALAppB2C.MSAL
+{
+    /// <summary>
+    /// Contains methods that initialize and use the MSAL SDK
+    /// </summary>
+    /// <autogeneratedoc />
+    public class MSALClientHelper
+    {
+        /// <summary>
+        /// As for the Tenant, you can use a name as obtained from the azure portal, e.g. kko365.onmicrosoft.com"
+        /// </summary>
+        private readonly AzureADB2CConfig _azureADB2CConfig;
+
+        /// <summary>
+        /// The public client application used by the app
+        /// </summary>
+        private IPublicClientApplication _publicClientApplication;
+        public IPublicClientApplication PublicClientApplication {
+            get
+            {
+                if (_publicClientApplication is null)
+                {
+                    _publicClientApplication = PublicClientApplicationBuilder.Build();
+                }
+
+                return _publicClientApplication;
+            }
+        }
+
+        private PublicClientApplicationBuilder _publicClientApplicationBuilder;
+        private PublicClientApplicationBuilder PublicClientApplicationBuilder
+        {
+            get
+            {
+                if (_publicClientApplicationBuilder is null)
+                {
+                    _publicClientApplicationBuilder = PublicClientApplicationBuilder.Create(_azureADB2CConfig.ClientId)
+                        .WithB2CAuthority($"{_azureADB2CConfig.Instance}/tfp/{_azureADB2CConfig.Domain}/{_azureADB2CConfig.SignUpSignInPolicyid}")
```

Comment:
>                         .WithB2CAuthority($"{_azureADB2CConfig.Instance}/tfp/{_azureADB2CConfig.Domain}/{_azureADB2CConfig.SignUpSignInPolicyid}")

`/tfp` is probably not needed here as going forward we will only use `xxx.b2clogin.com` format as instance